### ISSUE TITLE
Update failed due to Rosetta Installtion (not installed)

### ISSUE THREAD

**Issue by gcg** [docker/for-mac#7253]
### Description

Docker update (from v4.29.0) fails the update with following error: 

<img width="618" alt="image" src="https://github.com/docker/for-mac/assets/811685/b2ddc3a6-1370-485a-b22b-b61142eab8e0">

Clicking disable rosetta does nothing, the error window closes and reopens almost immediately. 

I never installed rosetta and don't plan on installing from this point on since all the images I use locally are arms.

Currently docker became unusable for me, I will try nuking it by removing everything and trying fresh install but this is crappy as hell as you guys can imagine. 

### Reproduce

1. click update and restart docker from system tray on a machine that has docker but not rosetta
2. you should see the error window with installing Rosetta: exit status 42 message. 
3. nothing from this point on works.

### Expected behavior

docker macos app updating and restarting it as it should. 

### docker version

```bash
Version 4.29.0 (145265)

Engine: 26.0.0

Compose: v2.26.1-desktop.1

Credential Helper: v0.8.1

Kubernetes: v1.29.2

go 1.22 

(docker is unresponsive so docker version doesn't respond.)
```

### docker info

```bash
N/A
```

### Diagnostics ID

2555A9B8-052C-460B-AA40-EF57FA21B996/20240416110811

### Additional Info

I couldn't gather  docker info since docker is unresponsive

**Comment by MihaelaStoica** [comment]
@gcg, we acknowledged this as a bug in 4.29.0 (see https://github.com/docker/for-mac/issues/7243) and we currently offer a development build with a fix: https://github.com/docker/for-mac/issues/7243#issuecomment-2045838577

**Comment by gcg** [comment]
> @gcg, we acknowledged this as a bug in 4.29.0 (see #7243) and we currently offer a development build with a fix: [#7243 (comment)](https://github.com/docker/for-mac/issues/7243#issuecomment-2045838577)

Awesome, thank you, I thought that was the case and searched for the issue but somehow I couldn't find it before I opened mine. 

I am closing it since we already have an issue for this and will try to switch to dev build as u suggested. 

cheers <3
